### ISSUE TITLE
Nonzero eigenstrain in zeroth timestep improperly added to total strain

### ISSUE THREAD

**Issue by sapitts** [idaholab/moose#8909]
### Description of the enhancement or error report
Eigenstrains which have a non-zero value at the start of the simulation (e.g. a non zero delta temperature for thermal expansion eigenstrains) do not correctly report the eigen strain in cases when the `incremental_form` option is not used or is not available.  The eigenstrain is calculated at the zeroth timestep, but the value of the eigenstrain is added to the mechanical strain instead of the total strain in the first timesteps; when the eigen strain is added to the total strain in the second timestep, the value of the total strain does not match the value of the eigenstrain:

### Rationale for the enhancement or information for reproducing the error
Use this modification of a regression test to reproduce the results below: 
[constant_expansion_coeff.i.zip](https://github.com/idaholab/moose/files/914252/constant_expansion_coeff.i.zip)

![screen shot 2017-04-11 at 11 57 47](https://cloud.githubusercontent.com/assets/9342949/24923286/486be43e-1eae-11e7-9210-b07c56e4ef93.png)

### Identified impact
There are at least a couple of paths forward to correct this:
  1. Bring back the incremental form of the eigenstrain in a base class, or
  2. Add in a `_step_zero` variable and check in the `ComputeEigenstrainBase.C` to prevent the eigenstrains from running in the first timestep (and thereby incorrectly contributing to the mechanical strain)
```
  void
  ComputeEigenstrainBase::computeQpProperties()
  {
    if (_t_step >= 1)
      _step_zero = false;
    if (!_step_zero)
      computeQpEigenstrain();
  }
```

A change is required to the eigenstrain calculations to ensure that we are correctly accounting for thermal strain, particularly in Bison runs where the `stress_free_temperature` and the initial temperature values are quite different.  @bwspenc let's discuss what option makes the most sense.

**Comment by bwspenc** [comment]
@sapitts, I've been testing your model, and I get the same behavior regardless of whether `incremental_form=true` or `false`.  At step zero, no solution is done, which explains why the mechanical strain is nonzero in that step.  The model never has a chance to adjust the displacements to put it in equilibrium as it does in the later steps.  I'm not sure what the consequences of having nonzero elastic strains in the zero step are.  I'm going to look into that now.

**Comment by bwspenc** [comment]
I see -- mechanical_strain is used all over the place, and it's stateful, so if it gets set to anything nonzero in the zero step, it messes everything up. I think your fix is the way to go.

**Comment by sapitts** [comment]
Closed by #8910
